### PR TITLE
qa/workunits/mon/crush_ops: some fixes for classes

### PR DESCRIPTION
**Pull request by liewegas**
Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

File: qa/workunits/mon/crush_ops.sh [hunk 1]
```diff
@@ -21,8 +21,12 @@ ceph osd crush rule create-simple bar default host
 
 # make sure we're at luminous+ before using crush device classes
 ceph osd require-osd-release luminous
-ceph osd crush set-device-class ssd osd.0
-ceph osd crush set-device-class hdd osd.1
+o0=`ceph osd create`
+ceph osd crush set-device-class ssd $o0
+expect_false ceph osd crush set-device-class hdd $o0
+ceph osd crush rm-device-class $o0
```
Comment (xiexingguo) [inline qa/workunits/mon/crush_ops.sh]: Actually this will not work because `rm-device-class` will automatically recycle empty class..

Comment (liewegas) [inline qa/workunits/mon/crush_ops.sh]: how about this?

**Comment by xiexingguo** [comment]
This is fragile. Actually there are other bugs preventing the `crush_ops.sh` script from working.
Can you try https://github.com/ceph/ceph/pull/16698 instead?

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-08-02_10:30:38-rados-wip-kefu-testing-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2017-08-01_17:41:46-rados-wip-kefu-testing-distro-basic-mira/

crush_ops.sh still fails.

**Comment by liewegas** [comment]
<pre>2017-08-03T06:43:02.627 INFO:tasks.workunit.client.0.smithi029.stderr:+ ceph osd pool create ec-foo 2 erasure ec-foo-ssd
2017-08-03T06:43:02.848 INFO:tasks.workunit.client.0.smithi029.stderr:2017-08-03 06:43:02.731108 7feab7816700 -1 WARNING: all dangerous and experimental features are enabled.
2017-08-03T06:43:02.849 INFO:tasks.workunit.client.0.smithi029.stderr:2017-08-03 06:43:02.782615 7feab7816700 -1 WARNING: all dangerous and experimental features are enabled.
2017-08-03T06:43:02.849 INFO:tasks.workunit.client.0.smithi029.stderr:2017-08-03 06:43:02.784364 7feaa5ffb700  0 -- - >> 172.21.15.149:6790/0 pipe(0x7feab00c6150 sd=7 :0 s=1 pgs=0 cs=0 l=1 c=0x7feab00c4db0).fault
2017-08-03T06:43:02.968 INFO:tasks.workunit.client.0.smithi029.stderr:Error EINVAL: device class ssd does not exist
</pre>
/a/sage-2017-08-02_18:54:18-rados-wip-sage-testing-distro-basic-smithi/1475158

**Pull request closed without merging**